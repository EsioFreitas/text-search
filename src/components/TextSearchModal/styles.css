.backdrop-open {
  position: fixed;
  width: 100vw;
  height: 100vh;
  background-color: #35353550;
  display: flex;
  justify-content: center;
  align-items: center;
  top: 0;
  left: 0;
}

.modal-open {
  background-color: white;
  padding: 1rem;
  border-radius: 0.5rem;
  box-shadow: 0 0 5px #35353510;
}

.backdrop-close {
  height: 0;
}

.modal-close {
  display: none;
}
