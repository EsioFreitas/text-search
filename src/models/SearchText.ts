export interface SearchText {
  id: string;
  title?: string ;
  status: string;
  urls: Array<string>;
}