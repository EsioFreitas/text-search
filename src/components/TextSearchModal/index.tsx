import React from "react";
import { SearchText } from "../../models/SearchText";
import "./styles.css";

interface TextSearchModalProps {
  searchText: SearchText;
}

const TextSearchModal = ({ searchText }: TextSearchModalProps) => {
  const renderStatusColor = (status: string) => {
    switch (status) {
      case "done":
        return "#4cb919";
      case "active":
        return "#b9af19";
      default:
        return "#ccc";
    }
  };

  const statusColor = renderStatusColor(searchText.status);
  return (
    <div className="search-card">
      <h3>{searchText.id}</h3>
      <div>
        <div className="search-status">
          <div style={{ backgroundColor: statusColor }} />
          <span style={{ color: statusColor }}>{searchText.status}</span>
        </div>
        <button>View</button>
      </div>
    </div>
  );
};

export default TextSearchModal;
