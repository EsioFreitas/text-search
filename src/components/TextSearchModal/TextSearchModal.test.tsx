import React from "react";
import { render } from "@testing-library/react";
import TextSearchModal from ".";
import { SearchText } from "../../models/SearchText";

describe("TextSearchCard tests", () => {
  const searchText: SearchText = {
    id: "asdew",
    title: "efadasdas",
    status: "active",
    urls: [],
  };
  it("should TextSearchCard component return", () => {
    const { baseElement } = render(
      <TextSearchModal
        isOpen={false}
        searchText={searchText}
        toggle={() => {}}
      />
    );
    expect(baseElement).toBeTruthy();
  });

  it("should render TextSearchCard component", async () => {
    const { container } = render(
      <TextSearchModal
        isOpen={false}
        searchText={searchText}
        toggle={() => {}}
      />
    );
    expect(container).toMatchSnapshot();
  });
});
