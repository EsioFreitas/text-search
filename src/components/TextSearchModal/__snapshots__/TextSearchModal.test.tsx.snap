// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextSearchCard tests should render TextSearchCard component 1`] = `
<div>
  <div
    class="backdrop-close"
  >
    <div
      class="modal-close"
    >
      <h3
        class="modal-title"
      >
        efadasdas
      </h3>
      <h4
        class="modal-subtitle"
        style="color: rgb(185, 175, 25);"
      >
        active
      </h4>
      <div
        class="image-container image-container-card"
      >
        <img
          alt="void"
          src="void.svg"
        />
      </div>
    </div>
  </div>
</div>
`;
